cub/cub/thread/thread_load.cuh

```diff
@@ -144,6 +144,39 @@ struct IterateThreadLoad<MAX, MAX>
   {}
 };
 
+namespace detail
+{
+template <CacheLoadModifier MODIFIER, typename T, int... Is>
+_CCCL_DEVICE _CCCL_FORCEINLINE void
+UnrolledThreadLoadImpl(T const* ptr, T* vals, ::cuda::std::integer_sequence<int, Is...>)
+{
+  // TODO(bgruber): replace by fold over comma in C++17
+  int dummy[]{(vals[Is] = ThreadLoad<MODIFIER>(ptr + Is), 0)...};
+  (void) dummy;
+}
+
+template <typename InputIteratorT, typename T, int... Is>
+_CCCL_DEVICE _CCCL_FORCEINLINE void
+UnrolledCopyImpl(InputIteratorT itr, T* vals, ::cuda::std::integer_sequence<int, Is...>)
```

Comment:
Important: In C++ an input_iterator has a well defined meaning. Please use a different name or `RandomAccessIteratorT`

Reply:
I know. Unfortunately, CUB uses the term "input iterator" a lot to mean an iterator to some input data, and then proceeds to subscript it like a random access iterator. I took the name "InputIteratorT" from the code I am replacing (`IterateThreadLoad`). But I will gladly call this a random iterator, because that's what it is!

Reply:
I am in favour of keeping it as is to stay consistent and have an issue to clean it up

Reply:
Alright, I will revert my latest changes then.

Reply:
Here is the issue: #1866